pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -1569,6 +1590,63 @@ private void internalDeleteSubscriptionForNonPartitionedTopic(AsyncResponse asyn
             });
     }
 
+    private void internalAnalyzeSubscriptionBacklogForNonPartitionedTopic(AsyncResponse asyncResponse,
```

Comment:
It's better to return CompletableFuture<AnalyzeSubscriptionBacklogResult> here to avoid passing the AsyncResponse to the internal of implement, it will help us to understand the returned type without looking at the implementation details

Reply:
We have changed most of the admin implements with future returned types.